### DISCUSSION TITLE
Pipewire portal grabber error on web client login (Latest beta version - MS2130)

### DISCUSSION THREAD

**Discussion by stickmanbrad** [awawa-dev/HyperHDR#623] [General]
#### Steps to reproduce

Booting up Raspberry PI and loading the HyperHDR web client as normal. Using the beta version because I’m using one of the few MS2130’s

#### What is expected?

Standard login procedure

#### What is actually happening?

Pop-up message with the following:

2023-08-13T17:01:50.179Z [PIPEWIRE_SYSTEM:AUTO] Could not load Pipewire proxy library. Error: libpipewire-0.3.so.0: cannot open shared object file: No such file or directory
2023-08-13T17:01:50.179Z [DAEMON] The system doesn't support the Pipewire/Portal grabber
2023-08-13T17:01:50.180Z [X11_SYSTEM:AUTO] Cropping image: width=0 height=0; crop: left=0 right=0 top=0 bottom=0 
2023-08-13T17:01:50.182Z [X11_SYSTEM:AUTO] Loaded X11 proxy library for screen capturing
2023-08-13T17:01:50.186Z [DAEMON] The system doesn't support the X11 grabber

#### System
<!-- In the web interface of the HyperHDR config go to System > About and Paste the content of "System info" here->

System info

HyperHDR Server: 
- Build:           fix_qt_5_15_for_Windows (Awawa-c4a8bf6/e010903-1691699464) **(AM I RUNNING THE WRONG VERSION FOR RASPBERRY PI?)**
- Build time:      Aug 10 2023 20:36:21
- Git Remote:      https://github.com/awawa-dev/HyperHDR
- Version:         20.0.0.0beta0
- UI Lang:         auto (BrowserLang: en-GB)
- UI Access:       default
- Avail Capt:      Linux (V4L2)
- Database:        read/write

HyperHDR Server OS: 
- Distribution:   Debian GNU/Linux 11 (bullseye)
- Architecture:   arm64
- CPU Type:       Raspberry Pi 4 Model B Rev 1.5
- CPU Revision:   d03115
- CPU Hardware:   BCM2835
- Kernel:         linux (6.1.21-v8+ (WS: 64))
- Qt Version:     5.15.2
- Browser:        Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.6 Safari/605.1.15

**Comment by stickmanbrad** [comment]
Adding system details below as I’m not sure if it’s cut it off:

System info

HyperHDR Server: 
- Build:           fix_qt_5_15_for_Windows (Awawa-c4a8bf6/e010903-1691699464) **(AM I RUNNING THE WRONG VERSION FOR RASPBERRY PI?)**
- Build time:      Aug 10 2023 20:36:21
- Git Remote:      https://github.com/awawa-dev/HyperHDR
- Version:         20.0.0.0beta0
- UI Lang:         auto (BrowserLang: en-GB)
- UI Access:       default
- Avail Capt:      Linux (V4L2)
- Database:        read/write

HyperHDR Server OS: 
- Distribution:   Debian GNU/Linux 11 (bullseye)
- Architecture:   arm64
- CPU Type:       Raspberry Pi 4 Model B Rev 1.5
- CPU Revision:   d03115
- CPU Hardware:   BCM2835
- Kernel:         linux (6.1.21-v8+ (WS: 64))
- Qt Version:     5.15.2
- Browser:        Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.6 Safari/605.1.15

**Comment by stickmanbrad** [comment]
Full Logs https://pastebin.com/TGi0YWgK

**Comment by awawa-dev** [comment]
Hi You are using right test version. Since lite/console version of Raspberry Pi OS lacks x-server and pipewire ecosystem you can ignore this error: HyperHDR simply bypass pipewire activation if it can't find it. Pipewire is used for the software screen capturing mainly for Wayland.

**Comment by stickmanbrad** [comment]
I also wanted to add that now I’m using the MS2130 capture device, do any of the settings in the attached image need to be changed?

[https://imgur.com/a/iScUvSa](url)

But the main thing is that Pipewire error I don’t understand.

Kind Regards.

**Comment by stickmanbrad** [comment]
> Hi You are using right test version. Since lite/console version of Raspberry Pi OS lacks x-server and pipewire ecosystem you can ignore this error: HyperHDR simply bypass pipewire activation if it can't find it. Pipewire is used for the software screen capturing mainly for Wayland.

Thank you for getting back to me so quickly! That’s good to know. I shall safely ignore this. Can I just ask about the above settings in the Imgur link? I’ve never really known if they were correct, but now I’m using this new capture card along with the correct LUT table, just wanted to know if I needed to change anything. I also soldered the connections instead of using those silly clips so hopefully that’ll help lol.

Kind Regards in advanced.

**Comment by awawa-dev** [comment]
Rather minor issue: anti-flickering threshold is too high. Default is 32 and anti-flickering filter is disabled above that value. On the video source (usually grabber) RGB is in range 0-255 and can be increased +/-1: so +1 for 10 on input it about 10%, +/- 1 for 32 is 3% and +/-1 for 200 is 0.5% It is a percentage increase/decrease in brightness in a great simplification at the smallest step (+/-1) depending on the initial value, the greater in percentage the greater the risk of a noticeable flicker.

**Comment by stickmanbrad** [comment]
> Rather minor issue: anti-flickering threshold is too high. Default is 32 and anti-flickering filter is disabled above that value. On the video source (usually grabber) RGB is in range 0-255 and can be increased +/-1: so +1 for 10 on input it about 10%, +/- 1 for 32 is 3% and +/-1 for 200 is 0.5% It is a percentage increase/decrease in brightness in a great simplification at the smallest step (+/-1) depending on the initial value, the greater in percentage the greater the risk of a noticeable flicker.

Thank you for the response. What about the ‘Backlight Threshold’? It’s currently set to 2. It does seem when there’s a black screen now between content, there is a very faint white glow. I suspect this has something to do with it?

The other thing I was going to ask. Now that I’m using an HDMI source splitter, I’ve noticed that when I turn the TV and video content source off, the LED device stays on now. I assume this is likely to do with the addition of the splitter inline. I have noticed a new option in the ‘Video Capturing’ tab called ‘Automatic signal detection’. It’s current disabled. Could this assist me at all?

Kind regards

**Comment by awawa-dev** [comment]
If the is no signal source (no grabber, no effect etc) the LEDs will be turned off completely. But if the is a signal (even a black board produces by the grabber when no HDMI signal)  ‘Backlight Threshold’ will prevent LEDs from turning off so you may want to change it to 0 if you like. Sometimes grabbers produce 'rainbow' or 'almost' black no signal (when no HDMI signal)  boards that won't turn LEDs off. Then you may want to use  ‘Automatic signal detection' that will learn HyperHDR to recognize such boards and turn off the LEDs (it was introduced some time ago in v17, there is more information on the blog related to that release).

**Comment by stickmanbrad** [comment]
> If the is no signal source (no grabber, no effect etc) the LEDs will be turned off completely. But if the is a signal (even a black board produces by the grabber when no HDMI signal) ‘Backlight Threshold’ will prevent LEDs from turning off so you may want to change it to 0 if you like. Sometimes grabbers produce 'rainbow' or 'almost' black no signal (when no HDMI signal) boards that won't turn LEDs off. Then you may want to use ‘Automatic signal detection' that will learn HyperHDR to recognize such boards and turn off the LEDs (it was introduced some time ago in v17, there is more information on the blog related to that release).

My main issue at the moment, is that there is a slight delay in lip syncing audio when people are talking on the video source. It’s slight but noticeable. I have no idea why it might be doing this. I’m using the Ezcootech ‎EZ-SP12H2 model to split the signal. The EDID switch is set to ‘COPY1’. Not sure if that’s correct. Source to capture card is downscaled to 1080p from 4K.

should Dolby vision be on on the source or just HDR? - I don’t think that’s related though.

kind regards

**Comment by stickmanbrad** [comment]
Oh and the audio goes to a Sonos Beam which is connected to the HDMI arc port on the Tv. I know arc isn’t supported on the splitter, but it’s directly connected to tv.

**Comment by awawa-dev** [comment]
The HDMI switch should not interfere with the sound. Either your video player generates unsynchronized audio or your TV while separating the audio from the video source or external speakers.

**Reply by stickmanbrad** [reply]
I have a splitter which has hdmi ports for the different devices prior to that hdmi splitter. Could that be causing the issues? There is definitely a delay in audio when looking at lip syncing. I’m not sure what to try because I didn’t have this issue before :/

**Comment by stickmanbrad** [comment]
> The HDMI switch should not interfere with the sound. Either your video player generates unsynchronized audio or your TV while separating the audio from the video source or external speakers.

Could it be that the EZ-SP12H2 isn’t feature rich enough to support the ARC pass through to the SONOS Beam through the TV? The lip sync is really annoying
